virtcontainers/clh.go

```diff
@@ -425,8 +432,62 @@ func (clh *cloudHypervisor) hypervisorConfig() HypervisorConfig {
 }
 
 func (clh *cloudHypervisor) resizeMemory(reqMemMB uint32, memoryBlockSizeMB uint32, probe bool) (uint32, memoryDevice, error) {
-	clh.Logger().WithField("function", "resizeMemory").Warn("not supported")
-	return 0, memoryDevice{}, nil
+	memDev := memoryDevice{probe: probe}
+	if reqMemMB == 0 {
+		return 0, memDev, errors.New("Can not resize memory to 0")
+	}
+
+	info, err := clh.vmInfo()
+	if err != nil {
+		return 0, memDev, err
+	}
+
+	currentMem := utils.MemUnit(info.Config.Memory.Size) * utils.Byte
+	newMem := utils.MemUnit(reqMemMB) * utils.MiB
+
+	// Early check to verify if boot memory is the same as requested
+	if currentMem == newMem {
+		clh.Logger().WithField("memory", reqMemMB).Debugf("VM already has requested memory")
+		return 0, memDev, nil
+	}
+
+	if currentMem > newMem {
+		clh.Logger().Warn("Remove memory is not supported, nothing to do")
+		return 0, memDev, nil
+	}
+
+	blockSize := utils.MemUnit(memoryBlockSizeMB) * utils.MiB
```

Comment:
Two questions:
1. Do we need to respect the `memoryBlockSizeMB`? It sounds only related when `memory device` hotplug is involved, and `qemu.go` does not check on it when `virtio-mem` is enabled.
2. Do we want to make sure we respect the constraint described in CLH hotplug [doc](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/docs/hotplug.md): `Due to guest OS limitations is is necessary to ensure that amount of memory added (between currently assigned RAM and that which is desired) is a multiple of 128MiB.`

Reply:
> Two questions:
> 
>     1. Do we need to respect the `memoryBlockSizeMB`? It sounds only related when `memory device` hotplug is involved, and `qemu.go` does not check on it when `virtio-mem` is enabled.

Yes the way the hotplug is without virtio-mem, so we need to verify we request to add memory as a multiple of 128MiB.

> 
>     2. Do we want to make sure we respect the constraint described in CLH hotplug [doc](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/docs/hotplug.md): `Due to guest OS limitations is is necessary to ensure that amount of memory added (between currently assigned RAM and that which is desired) is a multiple of 128MiB.`


Yes, the block size in x86_64 is set to 128Mb and the function `func (m MemUnit) AlignMem(blockSize MemUnit) MemUnit {` will make sure to make it a multiple of it